rmgpy/pdep/cse.pyx

```diff
@@ -223,3 +223,163 @@ def apply_chemically_significant_eigenvalues_method(network, list lumping_order=
+
+    k = np.zeros((n_isom + n_reac + n_prod, n_isom + n_reac + n_prod), np.float64)
+
+    # Extract the chemically-significant eigenvalues and eigenvectors
+    omega = np.real(omega0.take(ind[:]))
+
+    cse_omega = np.real(omega0.take(ind[:n_cse]))
+    eigen_vectors = np.real(eigen_vectors0.take(ind[:], axis=1))
+
+    for j in range(eigen_vectors.shape[1]):
+        eigen_vectors[:, j] *= s_mat
+        eigen_vectors[:, j] /= np.linalg.norm(eigen_vectors[:, j])
+
+    cse_eigen_vectors = eigen_vectors[:,:n_cse]
+
+    bi = np.zeros((n_reac,n_grains,n_j))
+    biunnormed = np.zeros((n_reac,n_grains,n_j))
+    for n in range(n_reac):
+        for eind in range(n_grains):
+            for jind in range(n_j):
+                bi[n,eind,jind] = dens_states[n+n_isom,eind,jind]*(2 * j_list[jind] + 1) * exp(-e_list[eind] * beta)
+
+    biunnormed[:,:,:] = bi[:,:,:]
+
+    for n in range(n_reac):
+        bi[n,:,:] /= np.sum(bi[n,:,:]) #normalize
+
+
+    Pv = np.zeros((me_mat.shape[0],n_reac))
+    k2v = np.zeros((me_mat.shape[0],n_reac))
+    for n in range(n_reac):
+        for i in range(n_isom):
+            for eind in range(n_grains):
+                for jind in range(n_j):
+                    index = indices[i,eind,jind]
+                    if index > -1:
+                        Pv[index,n] += f_im[i,n,eind,jind]*bi[n,eind,jind]
+                        k2v[index,n] += g_nj[n,i,eind,jind]
+
+
+    pvlmbd = np.linalg.solve(eigen_vectors,Pv)
+    Gvlmbd = eigen_vectors.T @ k2v
+
+    W = np.zeros((n_isom,n_isom))
+    for k in range(n_chem):
+        W[:,k] = states_to_configurations(network,indices,cse_eigen_vectors[:,k],exclude_association=True)
+
+    Winv = np.linalg.inv(W)
+
+
+    transformed_omega = W @ sparse.diags(omega[:n_chem]) @ Winv
+
+    kmat = np.zeros((n_channels,n_channels))
+
+    for i in range(n_isom):
+        for j in range(n_isom):
+            kmat[i,j] = transformed_omega[i,j]
+
+    for i,react in enumerate(channels):
+        for j,prod in enumerate(channels):
+            if i == j:
+                continue
+            if len(react.species) == 2 and len(prod.species) == 2:
+                kmat[j,i] = 0.0
+                for k,lmbd in enumerate(omega[n_cse:]):
+                    kmat[j,i] += -pvlmbd[k+n_cse,i-n_isom]*Gvlmbd[k+n_cse,j-n_isom]/lmbd
+
+            elif len(react.species) == 2:
+                kmat[j,i] = 0.0
+                kmat[j,i] = (W @ pvlmbd[:n_cse,i-n_isom])[j]
+
+            elif len(prod.species) == 2:
+                kmat[j,i] = 0.0
+                kmat[j,i] = (Gvlmbd[:n_cse,j-n_isom] @ Winv[:,i])
+
+    return kmat
+
+def apply_chemically_significant_eigenvalues_method_georgievskii(network, neglect_high_energy_collisions=neglect_high_energy_collisions,
+                                                                 high_energy_rate_tol=high_energy_rate_tol):
+    return get_rate_coefficients_CSE_Advanced(network, network.T, network.P, neglect_high_energy_collisions=False,
```

Comment:
Should use the variables instead of hard-coded?

Reply:
fixed